content/tutorials/Super-lightweight-realtime-subscriptions-with-SSE-and-Ably.textile

```diff
@@ -0,0 +1,223 @@
+        channel.publish('event', "Current date is %s" % (currentDT))
+        newChannel.publish('event', "Current date is %s" % (currentDT))
+        Timer(2.0, sendEvent).start()
+    Timer(2.0, sendEvent).start()
+    ```
+
+blang[php].
+    ```[php]
+    $client = new Ably\AblyRest('INSERT-YOUR-API-KEY-HERE');
+    $channel = $client->channel('example-channel');
+    $newChannel = $client->channel('new-channel');
+
+    function setInterval($func = null, $interval = 0){
+    if( ($func == null) || (!function_exists($func)) ){
+        throw new Exception('We need a valid function.');
+    }
+    $seconds = $interval * 1000;
+        while(true){
+            call_user_func($func); // Call the function you've defined.
+            usleep( $seconds );
+        }
+    }
+
+    function sendEvent(){
+    $date = new \DateTime('now');
+    $channel->publish('myEvent', "Current date is $date->format('D M d, Y G:i')" );
+    $newChannel->publish('myEvent', "Current date is $date->format('D M d, Y G:i')" );
+    }
+    setInterval('sendEvent', 2000);
+    ```
+
+h2(#setting-up-the-subscriber-with-http-event-streaming). Setting up the subscriber with HTTP Event Streaming
+
+Let us add the code below to our HTML file:
+
+```[html]
+<script type="text/javascript">
+
+const apiKey = "INSERT-YOUR-API-KEY-HERE";
+
+// can have multiple channels, comma-separated.
+// If a channel name is not url-safe, URL-encode it first
+const channels = "example-channel,new-channel";
+
+if(apiKey.indexOf("INSERT") === 0) {
+  alert("Add your API key to example.html");
+  throw 'Cannot proceed without api key';
+}
+
+const url = `https://realtime.ably.io/event-stream?v=1.1&key=${apiKey}&channels=${channels}`
+const eventSource = new EventSource(url);
+eventSource.onopen = function() {
+  addEvent(`Connected and subscribed to channels: ${channels.split(',').join(', ')}`);
+}
+
+eventSource.onmessage = function(event) {
+  const message = JSON.parse(event.data);
+  addEvent(`Channel: ${message.channel} - Message: ${message.name} - ${message.data}`);
+}
+
+eventSource.onerror = function(error) {
+  if(error.data) {
+    addEvent(`Error: ${error.data}`);
+  } else {
+    addEvent('Error connecting to Ably');
+  }
+}
+
+const eventList = document.querySelector('ul#events');
+
+function addEvent(text) {
+  const newElement = document.createElement("li");
+  newElement.textContent = text;
+  eventList.appendChild(newElement);
+}
+</script>
+```
+
+Here, we defined the script section, where we defined the @apiKey@ variable, which should be replaced with your Ably API key. We forge ahead to define our Ably channel with a variable named channels. We've added a simple check to see if the API key was in fact replaced. Next, we defined a variable called @url@, which is a template string, feeding in our @apiKey@ and @channels@. Notice that the route of this URL is @/event-stream@, which means we want to listen to the HTTP event streams from the server.
```

Comment:
> Here, we defined the script section,

In this step, we have defined our JavaScript logic to connect to the endpoint and display the updates returned in the @ul@ element in the console that we set up earlier.

Reply:
> the @apiKey@ variable, which should be replaced with your Ably API key.

Note that the @apiKey@ variable should be replaced with your own Ably API key.